functional_tests/scylla_operator/test_functional.py

```diff
@@ -462,6 +462,27 @@ def test_scylla_operator_pods(db_cluster: ScyllaPodCluster):
         [pods_info['name'] for pods_info in scylla_operator_pods if pods_info['status'] != 'Running'])
     assert not not_running_pods, f'There are pods in state other than running: {not_running_pods}'
 
+    # Cover https://github.com/scylladb/scylla-operator/issues/408:
+    # Operator shouldn't run as StatefulSet, should be Deployment
+    pods = db_cluster.k8s_cluster.kubectl(f"get pods {scylla_operator_pods[0]['name']} -o yaml",
+                                          namespace=SCYLLA_OPERATOR_NAMESPACE)
+    for owner_reference in yaml.safe_load(pods.stdout)['metadata']['ownerReferences']:
+        if not owner_reference:
+            continue
+
+        rs_details = db_cluster.k8s_cluster.kubectl(f"get replicaset {owner_reference['name']} -o yaml",
```

Comment:
This is not exactly `ReplicaSet`. As mentioned in the previous review, it may also be `StatefulSet` and `DaemonSet`.
So, such naming is not correct.

Then, we should only check the `Kind` field here. No need to read it's owner info as was mentioned in the previous review.

Expected something like following:

```
assert owner_reference["Kind"] == "ReplicaSet", (
    f"Expected 'ReplicaSet' kind as owner, but got {owner_reference['Kind']}")
```